### PR TITLE
[Core] Using the Create method for CL on the read_materials_process

### PR DESCRIPTION
**Pull request by loumalouomega**
#2291 should be updated consistently

#2431 must be cleaned-> Remove the new  read_materials_process_with_create and revert the structural solver

### CODE REVIEW

**Review by AlejandroCornejo — APPROVED** [review]
thank you vicente

**Comment by philbucher** [comment]
so now you use the create method based on the `Parameters`?

**Comment by loumalouomega** [comment]
Yes, but if not defined uses the create from constitutive_law.cpp, which basically just does a the same we were doing until now

**Comment by loumalouomega** [comment]
https://github.com/KratosMultiphysics/Kratos/blob/ec6716fb97bfe92ba91038322815157336e4e600/kratos/sources/constitutive_law.cpp#L84

**Comment by AlejandroCornejo** [comment]
In fact now I'm going to register all the new CL with the corresponding template so I think that the create is no longer necessary...

**Comment by loumalouomega** [comment]
Well, maybe it is, remember that @marandra was planing to use the Create for a different pourpose

File: kratos/python_scripts/read_materials_process.py [hunk 1]
```diff
@@ -108,7 +108,18 @@ def _GetConstitutiveLaw(self, my_string):
...
+
+        if len(splitted) == 0:
+            raise Exception("Something wrong. Trying to split the string " + my_string)
+        if len(splitted) > 3:
+            raise Exception("Something wrong. String " + my_string + " has too many arguments")
+
+        cl_name = splitted[-1]
+        param["name"].SetString(cl_name)
+        cl = self._get_attribute(cl_name, KratosMultiphysics.KratosGlobals.GetConstitutiveLaw, "Constitutive Law")
+        return cl.Create(param)
```
Comment (philbucher) [inline kratos/python_scripts/read_materials_process.py:122]: For this to work properly the Create function has to either be implemented in every CL or (what you are using) by passing the name of the CLaw within the Parameters
=> [this Function is virtual](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/constitutive_law.h#L518), and I don't see why
I.e. if someone overrides this function and does not pass the name, the proposed mechanism will fail!
Bottom line: Is this virtual really needed?

Comment (loumalouomega) [reply kratos/python_scripts/read_materials_process.py:122]: If the Create is not defined in every CL will take the one from the base class, so I will clone a CL from the KratosComponents, the same behaviour we were using until now

Comment (loumalouomega) [reply kratos/python_scripts/read_materials_process.py:122]: If someone overrides the function and doesn't know what is doing ... what I am suppose to do?

**Comment by AlejandroCornejo** [comment]
@loumalouomega yeah I think that it's interesting to have this feature but for us now it's not necessary once I've filled the add_cons_laws with the huge amount of combinations.

**Comment by loumalouomega** [comment]
Ok, in any case we should proceed in order to follow the design agreed

File: kratos/python_scripts/read_materials_process.py (hunk 1 above)
Comment (philbucher) [reply kratos/python_scripts/read_materials_process.py:122]: Thats why you might remove the virtual, this way it wont happen...

Just pointing out, not blocking. You guys are working with the CLaws, I don't know what your plans are

**Review by philbucher — APPROVED** [review]

**Comment by AlejandroCornejo** [comment]
I've just talked with @RiccardoRossi and we are going to use the Create anyways to allow the user to provide the combination of yield+plastic_potential and generate the string to define the name of the registered CL

**Comment by marandra** [comment]
Just to mention that I am using Create already. It will be necessary to add more generality required for multiscale analysis.

**Comment by RiccardoRossi** [comment]
@philbucher the Create function is intended to be virtual, since it can actually do very complicated things internally. (think for example to the FE^2 case, im which every CL is actually a full mesh of Finite Elements)

we however implemented a fallback in the base class so to cover the simple cases and avoid breaking anything.

This design was approved by the @KratosMultiphysics/technical-committee , however let me look into the code before we merge. I'll try to do it asap

**Comment by RiccardoRossi** [comment]
@loumalouomega thank you for reading carefully what i write :-D . Cite: "let me look into the code before we merge. I'll try to do it asap"

having said this code looks ok, so i would have said to go ahead

**Comment by loumalouomega** [comment]
The comment was published in the momment I merged

**Comment by loumalouomega** [comment]
imagen
Sorry, as you can see you comment was published when I did the merge, and sencond later I deleted the branch
